src/Sentry.NLog/NLogDiagnosticLogger.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+
+using NLog.Common;
+using Sentry.Infrastructure;
+
+namespace Sentry.NLog
+{
+    internal class NLogDiagnosticLogger : IDiagnosticLogger
+    {
+        private readonly IDiagnosticLogger _extraLogger;
+
+        public NLogDiagnosticLogger(IDiagnosticLogger extraLogger = null)
+        {
+            if (!InternalLogger.LogToConsole || !(extraLogger is ConsoleDiagnosticLogger))
+            {
+                _extraLogger = extraLogger;
+            }
+        }
+
+        public bool IsEnabled(SentryLevel level)
+        {
+            if (_extraLogger?.IsEnabled(level)==true)
+                return true;
+
+            switch (level)
+            {
+                case SentryLevel.Fatal: return InternalLogger.IsFatalEnabled;
+                case SentryLevel.Error: return InternalLogger.IsErrorEnabled;
+                case SentryLevel.Warning: return InternalLogger.IsWarnEnabled;
+                case SentryLevel.Info: return InternalLogger.IsInfoEnabled;
+                default: return InternalLogger.IsDebugEnabled;
+            }
+        }
+
+        public void Log(SentryLevel logLevel, string message, Exception exception = null, params object[] args)
+        {
+            switch (logLevel)
```

Comment:
I believe you need to guard again for the log level here. Now one guard for each logger:

Before writing to `extraLogger` do the:

```
           if (_extraLogger?.IsEnabled(level) == true)
            {
                return true;
            }
```

Before writing to the `InternalLogger`, do the check:

```
            switch (level)
            {
                case SentryLevel.Fatal: return InternalLogger.IsFatalEnabled;
                case SentryLevel.Error: return InternalLogger.IsErrorEnabled;
                case SentryLevel.Warning: return InternalLogger.IsWarnEnabled;
                case SentryLevel.Info: return InternalLogger.IsInfoEnabled;
                default: return InternalLogger.IsDebugEnabled;
            }
```

That assumes they can have different log levels enabled and if either returns true on `IsEnabled` we'd get a call here to `Log`

Reply:
The NLog  InternalLogger-methods performs implicit LogLevel-filtering-check, so no need to do it twice.